packages/replay/metadata/source.ts

```diff
@@ -1,35 +1,94 @@
 import type { Struct } from "superstruct";
-const { create, object, optional, string } = require("superstruct");
+const { create, object, optional, defaulted } = require("superstruct");
 
+import { envString } from "./env";
 import { UnstructuredMetadata } from "./types";
 
+const defaultObject = (objStruct: any) => optional(defaulted(object(objStruct), {}));
+
 const VERSION = 1;
 
 const versions: Record<number, Struct> = {
   1: object({
```

Comment:
Woah this is cool! I'm just kinda taking it at face value that this does what I expect, but looks good to me.